File: classify-rest-webservice/src/main/java/io/ermdev/alice/repository/RoleRepository.java
package io.ermdev.alice.repository;

import io.ermdev.alice.entity.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface RoleRepository extends JpaRepository<Role, Long> {

    @Query("from Role where id=:roleId")
    Role findById(@Param("roleId") Long id);

    @Query("from Role")
    List<Role> findAll();
}

File: resource/src/main/kotlin/io/ermdev/classify/data/repository/RoleRepository.kt
package io.ermdev.classify.data.repository

import io.ermdev.classify.data.entity.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface RoleRepository : JpaRepository<Role, Long> {

    @Query("from Role where name=:name")
    fun findByName(@Param("name") name: String): Optional<Role>
}